apps/controllo-ore-x/src/app/modules/auth/modules/team/dialogs/team-upsert/team-upsert.page.ts

```diff
@@ -0,0 +1,128 @@
+import { Component } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import {
+  ApiPaginatedResponse,
+  RoleReadDto,
+  UserCreateDto,
+  UserReadDto,
+  UserUpdateDto,
+} from '@api-interfaces';
+import { RoleDataService } from '@app/_core/services/role-data.service';
+import { TeamDataService } from '@app/_core/services/team-data.service';
+import { UpsertPage } from '@app/_shared/classes/upsert-page.class';
+import {
+  SubscriptionsLifecycle,
+  completeSubscriptions,
+} from '@app/utils/subscriptions_lifecycle';
+import { AlertService } from 'libs/rt-shared/src/alert/services/alert.service';
+import { RtDialogService } from 'libs/rt-shared/src/rt-dialog/services/rt-dialog.service';
+import { RT_FORM_ERRORS, RtFormError } from 'libs/utils';
+import { Subscription } from 'rxjs';
+import { TeamFormHelper } from '../../helpers/team.form-helper';
+
+@Component({
+  selector: 'controllo-ore-x-team-upsert',
+  templateUrl: './team-upsert.page.html',
+  styleUrls: ['./team-upsert.page.scss'],
+  providers: [TeamFormHelper],
+})
+export class TeamUpsertPage
+  extends UpsertPage<UserReadDto, UserCreateDto, UserUpdateDto>
+  implements SubscriptionsLifecycle
```

Comment:
```suggestion
  implements SubscriptionsLifecycle, OnDestroy, OnInit
```
Better to have explicit interfaces